framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2204,12 +2212,70 @@ public ParameterizedExecutableType methodFromUse(
 
     if (tree.getKind() == Tree.Kind.METHOD_INVOCATION
         && TreeUtils.isMethodInvocation(tree, objectGetClass, processingEnv)) {
-      adaptGetClassReturnTypeToReceiver(methodType, receiverType);
+      adaptGetClassReturnTypeToReceiver(methodType, receiverType, tree);
     }
 
     return new ParameterizedExecutableType(methodType, typeargs);
   }
 
+  /**
+   * Apply capture conversion to the type arguments of a method invocation.
+   *
+   * @param typeVarMapping mapping from type variable in the method declaration to the corresponding
+   *     type argument at the method invocation
+   * @param declTypeVar list of type variable declarations
+   * @return a mapping from type variable in the method declaration to its captured type argument.
+   *     For a non-wildcard, Capture conversion is the identity. Its keys are the same as in {@code
+   *     typeVarMapping}, and the values are their captures (capture conversion may be the
+   *     identity).
+   */
+  private Map<TypeVariable, AnnotatedTypeMirror> captureMethodTypeArgs(
+      Map<TypeVariable, AnnotatedTypeMirror> typeVarMapping,
+      List<AnnotatedTypeVariable> declTypeVar) {
+    // TODO: This should happen as part of Java 8 inference and this method should be removed when
+    // #979 is fixed.
+    Map<TypeVariable, AnnotatedTypeVariable> typeParameter = new HashMap<>();
+    for (AnnotatedTypeVariable t : declTypeVar) {
+      typeParameter.put(t.getUnderlyingType(), t);
+    }
+    // `newTypeVarMapping` is the result of this method.
+    Map<TypeVariable, AnnotatedTypeMirror> newTypeVarMapping = new HashMap<>();
+    Map<TypeVariable, AnnotatedTypeVariable> capturedTypeVarMapping = new HashMap<>();
+    for (Map.Entry<TypeVariable, AnnotatedTypeMirror> entry : typeVarMapping.entrySet()) {
+      AnnotatedTypeMirror originalTypeArg = entry.getValue();
+      TypeVariable typeVariable = entry.getKey();
```

Comment:
As a matter of style, I think a Map.Entry loop body is clearer when the `getKey()` invocation is on the first line and the `getValue()` invocation is on the second line.